commands/vehicle.go

```diff
@@ -113,12 +113,22 @@ func RequestPIDRaw(unitID uuid.UUID, request models.PIDRequest) (hexResp []strin
 	if err != nil {
 		return
 	}
-	hexResp = strings.Split(resp.Value, "\n")
-	for i := range hexResp {
-		if len(hexResp[i]) > 0 && !isValidHex(hexResp[i]) {
-			err = fmt.Errorf("invalid return value: %s", hexResp[i])
-			return
+	switch v := resp.Value.(type) {
+	case string:
```

Comment:
i feel like we should change the hexResp, either rename it or return an object eg. ObdResponse{IsHex, ValueHex, ValueFloat}
idk, something to make it a bit cleaner as to what is being returned. Like if we go into the case float64: we know it will never be a hex string response but always a float

Reply:
Updated